### PR TITLE
style(vpassword focused): verify password focused when switching from sign in to sign up

### LINKED ISSUE #6040: Verify password not focused when switching from sign in to sign up
Switching from `unknown account . sign up?` has bad focus behaviour 

@shane-tomlinson what should this do? focus `vpassword` first?

### PR DESCRIPTION
**Pull request by farhan787**
fixes #6040

### CODE REVIEW

**Comment by farhan787** [comment]
@shane-tomlinson is it good now ?

File: app/scripts/views/sign_up.js [hunk 1]
```diff
@@ -34,7 +34,7 @@ define(function (require, exports, module) {
     } else if (! password) {
       return 'input[type=password]';
     } else {
-      return '#age';
```
Comment (shane-tomlinson) [inline app/scripts/views/sign_up.js]: We'll still want to go to #age if there is both a password and a vpassword

Comment (farhan787) [reply app/scripts/views/sign_up.js]: @shane-tomlinson  but how can there be vpassword, user has to manually type it while signing up or it should be prefilled like password field ?

Comment (vbudhram) [reply app/scripts/views/sign_up.js]: @farhan787 If you wanted to do this check you would probably need to update the function [here](https://github.com/mozilla/fxa-content-server/blob/881efbc9a626a167aaa4d76291ef0cdd3db99d78/app/scripts/views/sign_up.js#L91) to find the "#vpassword" element and then pass it along to this function where you can select appropriate element to focus.

**Comment by vbudhram** [comment]
@shane-tomlinson I noticed the linked bug is closed, is this still issue still applicable?

**Comment by shane-tomlinson** [comment]
> @shane-tomlinson I noticed the linked bug is closed, is this issue still applicable?

As I said above, I view it a bit as busywork since we are deprecating this flow, but if it fixes the problem then we can merge.

**Comment by farhan787** [comment]
@shane-tomlinson I want to ask you that if a user is redirected to sign up page with email and password auto-filled up then the vpassword field should be focused instead of age, and the user should retype his/her password to confirm the password. So, how come we want to focus the age field ?

**Comment by shane-tomlinson** [comment]
> So, how come we want to focus the age field ?

We want the vpassword field focused *if there is no vpassword pre-fileld*. If there is a vpassword prefilled, then we want age focused.

**Comment by farhan787** [comment]
> If there is a vpassword prefilled, then we want age focused.

@shane-tomlinson  that's what I want to ask, in which case we have vpassword prefilled ?

**Comment by shane-tomlinson** [comment]
> @shane-tomlinson that's what I want to ask, in which case we have vpassword prefilled ?

Ahha! Here's a video to show you:

vpassword-prefilled

Here's the flow:

1. Open https://127.0.0.1:3030?action=email
2. Enter a new username
3. On the password page, enter both a password and vpassword
4. Click "Mistyped email?"
5. On the "enter your email" screen, re-submit the original email.
6. Both password and vpassword are prefilled.

File: app/scripts/views/sign_up.js (hunk 1 above)
Comment (farhan787) [reply app/scripts/views/sign_up.js]: @vbudhram thanks for help, I was doing that but I was confused about why do we need to check for retype password(vpassword) field, but now @shane-tomlinson cleared my doubt.

**Comment by farhan787** [comment]
> Ahha! Here's a video to show you:
> 
> vpassword-prefilled

@shane-tomlinson I noticed in this video too, the autofocus isn't working good because it should focus on age field as password and vpassword are pre-filled and that's why I fixed it too. Please review it :-)

File: app/scripts/views/sign_up_password.js [hunk 2]
```diff
@@ -17,6 +17,16 @@ import ServiceMixin from './mixins/service-mixin';
 import SignUpMixin from './mixins/signup-mixin';
 import Template from 'templates/sign_up_password.mustache';
 
+function selectAutoFocusEl(password, vPassword) {
```
Comment (shane-tomlinson) [inline app/scripts/views/sign_up_password.js]: I didn't even think to add it here! Well spotted!

Comment (farhan787) [reply app/scripts/views/sign_up_password.js]: Thanks!

**Pull request merged**